### DISCUSSION TITLE
Prometheus pending

### DISCUSSION THREAD

**Discussion by mhausenblas** [open-telemetry/opentelemetry-demo#765] [Q&A] [answered]
I have a 3 nodes EKS cluster (with managed node group) and all the pods come up but the Prometheus server. There's no indication for why, enough resources are available. What am I missing?

**Comment by julianocosta89** [comment]
If I am not mistaken Prometheus has a Volume configured.
If you are not deploying an EBS, you can simply update the Prometheus deployment to use an emptyDir.

@puckpuck maybe we could set `emptyDir` as default. What do you think?

**Reply by fatsheep9146** [reply]
I agreed with `emptyDir` volume as default.

**Reply by mhausenblas** [reply]
Thanks! So, where's the storage class to configure then?

**Reply by joshleecreates** [reply]
I think you would need to set `storageClass` under `prometheus` in your custom values, see https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L377

**Comment by mhausenblas** [comment, marked as answer]
OK, thanks @joshleecreates and it seems it's in the [CRD](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-helm-charts%20storageclass&type=code). Is there an example anywhere or do I have to dig into that? :) 

UPDATE: here's the fix …

1. Create a file called `values.yaml` with the following content:

```yaml
prometheus:
  server:
    persistentVolume:
      enabled: false
```

2. Then execute:

```shell
$ helm upgrade my-otel-demo open-telemetry/opentelemetry-demo --values values.yaml
```

**Reply by joshleecreates** [reply]
It looks like there's actually just a flag to set it enabled/disabled. This values snippet works for me:
```yaml
prometheus:
  server:
    persistentVolume:
      enabled: false
```

**Reply by mhausenblas** [reply]
Thanks a lot and will give it a try!

**Reply by puckpuck** [reply]
This should be a default on the Helm chart. I'll do some quick testing to make sure no weird side effects (ie: increased memory)

**Reply by puckpuck** [reply]
This PR should do what we need: https://github.com/open-telemetry/opentelemetry-helm-charts/pull/661

**Reply by mhausenblas** [reply]
@joshleecreates can confirm that fixed it (with the content above in `values.yaml` and then a `helm upgrade my-otel-demo open-telemetry/opentelemetry-demo --values values.yaml`). Thank you so much, really appreciate it!

@puckpuck that's awesome, can't wait, thanks a bunch!

**Reply by julianocosta89** [reply]
Thanks everyone!
@joshleecreates I can't mark your reply as answer 😢

**Reply by mhausenblas** [reply]
No worries I already did ;)
